proto/buf/alpha/registry/v1alpha1/image.proto

```diff
@@ -34,3 +37,53 @@ message GetImageRequest {
 message GetImageResponse {
   buf.alpha.image.v1.Image image = 1;
 }
+
+enum PartialImageLevel {
+  PARTIAL_IMAGE_LEVEL_UNSPECIFIED = 0;
+
+  // Returns complete files in which the requested types are contained.
+  PARTIAL_IMAGE_LEVEL_FILES = 1;
+
+  // Returns trimmed files in which the requested types are contained. In a
+  // trimmed file, all declarations that are not embedded (directly or
+  // indirectly) in requested types are removed.
+  PARTIAL_IMAGE_LEVEL_TYPES = 2;
+}
+
+message GetPartialImageRequest {
+  string owner = 1;
+  string repository = 2;
+  string reference = 3;
+
+  // If the image should exclude depdendencies. When set to true, a incomplete
+  // image without descriptors for dependencies of the requested types is
+  // returned.
+  bool exclude_dependencies = 4;
+
+  // Determines if the patterns specified in {messages,enums,services} should
+  // consider types from imported modules.
+  bool match_dependencies = 5;
```

Comment:
Also a cool idea, but I wonder if we could drop this option for now. Based on my comment above, if we include the transitive closure of the target type, we would only need a single RPC and I don't think we'd need this option at all. 

Was the intention that a separate RPC would be issued for the `google.type.DateTime` definition against the same module reference (e.g. `buf.build/acme/petapis`)? Then both of the results would be used by request builder?

I could see the value of an option like this in general since it's not always clear where the `google.type.DateTime` type is defined, but I'm not sure it's necessary for this use case. I could be missing something though.

Reply:
Related to the point above, with dependencies here I mean to refer to dependencies as in buf modules (from buf.lock's).

---

The intention here is that when I query for or "services in buf module x" I don't want to match all services in its dependencies (say we import googleapis, I don't want to show googleapi's services). However, at the point that I would want to query for a partial image with method `y`'s input type there is no guarantee in which bsr module that type is. In most cases (especially when people follow our lint rules) it will be, but it isn't required. 

For this API to work for request builder, I'm looking for a way to express that when I query for services I only want matches in this buf module 'shallowly' (without file imports) whereas if I'm querying for the request/response I want types from any bsr module in the image 'deeply' (with file imports needed).